File: app/src/main/java/gardezi/io/amiibofinder/dagger/AppComponent.java
package gardezi.io.amiibofinder.dagger;

import javax.inject.Singleton;

import dagger.BindsInstance;
import dagger.Component;
import dagger.android.AndroidInjectionModule;
import gardezi.io.amiibofinder.AmiiboApplication;

@Singleton
@Component(modules = {AndroidInjectionModule.class, AppModule.class, MainActivityModule.class})
public interface AppComponent {

    @Component.Builder
    interface Builder {
        @BindsInstance
        Builder application(AmiiboApplication application);
        AppComponent build();
    }

    void inject(AmiiboApplication application);
}

File: app/src/main/java/gardezi/io/amiibofinder/dagger/AppComponent.kt
package gardezi.io.amiibofinder.dagger

import javax.inject.Singleton

import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjectionModule
import gardezi.io.amiibofinder.AmiiboApplication

@Singleton
@Component(modules = arrayOf(AndroidInjectionModule::class, AppModule::class, MainActivityModule::class))
interface AppComponent {

    @Component.Builder
    interface Builder {
        @BindsInstance
        fun application(application: AmiiboApplication): Builder

        fun build(): AppComponent
    }

    fun inject(application: AmiiboApplication)
}
